enhancements/compliance-operator.md

```diff
@@ -0,0 +1,222 @@
+   means we need a post-run gather operation.
+ * For viewing the results by administrator might use [a script](https://github.com/jhrozek/scapresults-k8s/blob/master/scapresults/fetchresults.py)
+
+### Risks and Mitigations
+ * [jhrozek]: Not sure what to write here. The scan might be
+   resource-intensive perhaps? The scanner containers need to mount the
+   host fs? The permissions need careful review?
+
+
+## Design Details
+
+The `compliance-operator` would use the following API to track a compliance scan status:
+
+### API Specification
+The proposed compliance scan API instance
+
+```
+apiVersion: complianceoperator.compliance.openshift.io/v1alpha1
+kind: ComplianceScan
+metadata:
+  name: example-scan
+spec:
+  profile: xccdf_org.ssgproject.content_profile_ospp
+  content: ssg-ocp4-ds.xml
+```
+
+The profiles and content would be provided as part of the container image. This
+way, we can ensure that the auditor is using officially supported content.
+
+## Need Feedback
+
+How we present the remediations and allow the cluster administrator to execute
+them is something that should be discussed more.
+
+On a high level, there will be three kinds of remediations:
+ * General free-form text guidances. For example, your IDP must support 2FA.
+ * More specific advise, but still needs to be applied manually. For example,
+   "configure a message of the day so that a legal notice gets displayed after
+   login by creating a ConfigMap called `motd` in the `openshift` namespace."
+ * Gaps that can be remediated in a completely autonomous fashion. For example
+   making sure that the `auditd` service is enabled.
+
+At first, we would like to display the remediation in the HTML report so that
+the administrator can copy the advise and, if it's possible to apply directly,
+do that.
+
+However, when it comes to applying the remediation, there are two options:
+
+* Output the remediations to yaml files that the user can download and apply.
+  The caveat of this is that it makes it cumbersome: First check the report,
+  then download the remediations, then apply them manually.
+
+* Add the capability so that the opreator can schedule a workload that will
+  apply these remediations. The workflow would be as follows: Check the report
+  then modify the CRD so that the remediation is applied. This is more
+  automatic and user-friendly, however, it requires us giving a lot of
+  privileges to the workload, which might not be ideal.
+
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+TBD
+
+
+### Version Skew Strategy
+
+TBD
+
+## Implementation History
+
+TBD
+
+## Drawbacks
+
+TBD
+
+## Alternatives
+
+Provide a "cookbook" of recipes that provide the cluster administrator
+with a list of things to check and configure so that their cluster
+is compliant.
```

Comment:
This doesn't fully address the compliance needs, nor the compliance checks that some NIST controls require. So... going for the operator or a way to run openscap on a cluster is really needed.

Reply:
OK, I'm dropping this section completely then.